src/scheduler/analysis.py

```diff
@@ -157,26 +161,23 @@ def update_analysis_of_object_and_children(self, fo: FileObject):
         self._check_further_process_or_complete(fo)
 
     def start_analysis_of_object(self, fo: FileObject):
-        '''
+        """
         This function is used to start analysis of a firmware object. The function registers the firmware with the
         status module such that the progress of the firmware and its included files is tracked.
 
         :param fo: The firmware that is to be analyzed
-        '''
-        if self.status.file_should_be_analyzed(fo):
-            self.status.add_to_current_analyses(fo)
-            self.task_scheduler.schedule_analysis_tasks(fo, fo.scheduled_analysis, mandatory=True)
-            self._check_further_process_or_complete(fo)
-        else:
-            logging.info(f'Skipping analysis of {fo.uid} (duplicate file)')
+        """
+        self.status.add_object(fo)
+        self.task_scheduler.schedule_analysis_tasks(fo, fo.scheduled_analysis, mandatory=True)
+        self._check_further_process_or_complete(fo)
```

Comment:
We no longer check whether files should be analyzed. I understand that the analysis status now uses sets, which appear to make `file_should_be_analyzed` obsolete. But what about `schedule_analysis_tasks`? It has not been modified. Are there any possible side effects of that missing duplicate check condition?

Reply:
`file_should_be_analyzed` was already obsolete and should have been removed with #1050